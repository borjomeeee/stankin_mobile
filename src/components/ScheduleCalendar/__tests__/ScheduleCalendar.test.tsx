import React from 'react';
import renderer from 'react-test-renderer';

import {shallow, ShallowWrapper} from 'enzyme';

import ScheduleCalendarComponent from '../ScheduleCalendar.component';

describe('Schedule calendar tests', () => {
  it('Schedule calendar render correctly', () => {
    const tree = renderer
      .create(
        <ScheduleCalendarComponent
          todayDate={new Date(2020, 1, 1)}
          currDate={new Date(2020, 1, 1)}
          setCurrDate={jest.fn()}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Schedule select render correctly', () => {
    const dateDay = 1;
    const date = new Date(2020, 1, dateDay);

    const wrapper = shallow(
      <ScheduleCalendarComponent
        todayDate={date}
        currDate={date}
        setCurrDate={jest.fn()}
      />,
    );

    const isSelectedWrappers = wrapper
      .find('Styled(Text)')
      .findWhere(
        (node: ShallowWrapper<any, any, React.Component<{}, {}, any>>) =>
          node.prop('isSelected') === true,
      );

    expect(isSelectedWrappers).toHaveLength(1);
    expect(isSelectedWrappers.text()).toBe(dateDay.toString());

    expect(
      wrapper.findWhere(
        (node: ShallowWrapper<any, any, React.Component<{}, {}, any>>) =>
          node.key() === date.getTime().toString(),
      ),
    ).toHaveLength(1);

    expect(
      wrapper
        .findWhere(
          (node: ShallowWrapper<any, any, React.Component<{}, {}, any>>) =>
            node.key() === date.getTime().toString(),
        )
        .find('Styled(Text)')
        .first()
        .text(),
    ).toBe('Сб');
  });
});
