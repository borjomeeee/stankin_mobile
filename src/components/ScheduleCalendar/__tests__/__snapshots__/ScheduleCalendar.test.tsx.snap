// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule calendar tests Schedule calendar render correctly 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#444444",
              "fontSize": 16,
            },
          ]
        }
      >
        Сегодня
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#444444",
              "fontSize": 16,
            },
          ]
        }
      >
        1 Февраля
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderTopColor": "#E4E4E4",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Inter-Regular",
              "fontSize": 14,
            },
          ]
        }
      >
        Пн
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 5,
            "display": "flex",
            "height": 24,
            "justifyContent": "center",
            "marginTop": 2,
            "opacity": 1,
            "width": 24,
          }
        }
      >
        <Text
          isSelected={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Inter-Regular",
                "fontSize": 14,
              },
            ]
          }
        >
          27
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Inter-Regular",
              "fontSize": 14,
            },
          ]
        }
      >
        Вт
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 5,
            "display": "flex",
            "height": 24,
            "justifyContent": "center",
            "marginTop": 2,
            "opacity": 1,
            "width": 24,
          }
        }
      >
        <Text
          isSelected={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Inter-Regular",
                "fontSize": 14,
              },
            ]
          }
        >
          28
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Inter-Regular",
              "fontSize": 14,
            },
          ]
        }
      >
        Ср
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 5,
            "display": "flex",
            "height": 24,
            "justifyContent": "center",
            "marginTop": 2,
            "opacity": 1,
            "width": 24,
          }
        }
      >
        <Text
          isSelected={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Inter-Regular",
                "fontSize": 14,
              },
            ]
          }
        >
          29
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Inter-Regular",
              "fontSize": 14,
            },
          ]
        }
      >
        Чт
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 5,
            "display": "flex",
            "height": 24,
            "justifyContent": "center",
            "marginTop": 2,
            "opacity": 1,
            "width": 24,
          }
        }
      >
        <Text
          isSelected={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Inter-Regular",
                "fontSize": 14,
              },
            ]
          }
        >
          30
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Inter-Regular",
              "fontSize": 14,
            },
          ]
        }
      >
        Пт
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 5,
            "display": "flex",
            "height": 24,
            "justifyContent": "center",
            "marginTop": 2,
            "opacity": 1,
            "width": 24,
          }
        }
      >
        <Text
          isSelected={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Inter-Regular",
                "fontSize": 14,
              },
            ]
          }
        >
          31
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Inter-Regular",
              "fontSize": 14,
            },
          ]
        }
      >
        Сб
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#444444",
            "borderRadius": 5,
            "display": "flex",
            "height": 24,
            "justifyContent": "center",
            "marginTop": 2,
            "opacity": 1,
            "width": 24,
          }
        }
      >
        <Text
          isSelected={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Inter-Regular",
                "fontSize": 14,
              },
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Inter-Regular",
              "fontSize": 14,
            },
          ]
        }
      >
        Вс
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 5,
            "display": "flex",
            "height": 24,
            "justifyContent": "center",
            "marginTop": 2,
            "opacity": 1,
            "width": 24,
          }
        }
      >
        <Text
          isSelected={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Inter-Regular",
                "fontSize": 14,
              },
            ]
          }
        >
          2
        </Text>
      </View>
    </View>
  </View>
</View>
`;
